lib/gollum-lib/markups.rb

```diff
@@ -6,7 +6,7 @@ class Markup
         Kramdown::Document.new(content, :auto_ids => false, :input => "markdown").to_html
     }
 
-    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/)
+    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/, :h1 => /^(#\s+)(\w+.*)/)
```

Comment:
> So since a Page's content is basically static, I think there needs to be a method that returns the a page's content, split into before, wanted section, and after. That array can then be used in the frontend to display only the wanted section, joining it again with before and after when the form is submitted.

Yes, that's another possibility :).